types/v56/types.go

```diff
@@ -2179,29 +2180,44 @@ type QueryResultCatalogItemType struct {
-	IsExpired          bool      `xml:"isExpired,attr,omitempty"`          // True if this entity is expired
-	StorageProfileName string    `xml:"storageProfileName,attr,omitempty"` // Storage profile name
-	Version            string    `xml:"version,attr,omitempty"`            // Storage profile name
-	LastSuccessfulSync string    `xml:"lastSuccessfulSync,attr,omitempty"` // Date of last successful sync
-	Link               *Link     `xml:"Link,omitempty"`
-	Metadata           *Metadata `xml:"Metadata,omitempty"`
+	HREF                   string    `xml:"href,attr,omitempty"`               // The URI of the entity.
+	ID                     string    `xml:"id,attr,omitempty"`                 // vApp template ID.
+	Type                   string    `xml:"type,attr,omitempty"`               // The MIME type of the entity.
+	OwnerName              string    `xml:"ownerName,attr,omitempty"`          // Owner name
+	CatalogName            string    `xml:"catalogName,attr,omitempty"`        // Catalog name
+	IsPublished            bool      `xml:"isPublished,attr,omitempty"`        // True if this entity is in a published catalog
+	Name                   string    `xml:"name,attr,omitempty"`               // vApp template name.
+	Description            string    `xml:"description,attr,omitempty"`        // vApp template description.
+	Vdc                    string    `xml:"vdc,attr,omitempty"`                // VDC reference or ID
+	VdcName                string    `xml:"vdcName,attr,omitempty"`            // VDC name
+	Org                    string    `xml:"org,attr,omitempty"`                // Organization reference or ID
+	CreationDate           string    `xml:"creationDate,attr,omitempty"`       // Creation date
+	IsBusy                 bool      `xml:"isBusy,attr,omitempty"`             // True if the vApp template is busy
+	IsGoldMaster           bool      `xml:"isGoldMaster,attr,omitempty"`       // True if the vApp template is a gold master
+	IsEnabled              bool      `xml:"isEnabled,attr,omitempty"`          // True if the vApp template is enabled
+	Status                 string    `xml:"status,attr,omitempty"`             // Status
+	IsDeployed             bool      `xml:"isDeployed,attr,omitempty"`         // True if this entity is deployed
+	IsExpired              bool      `xml:"isExpired,attr,omitempty"`          // True if this entity is expired
+	StorageProfileName     string    `xml:"storageProfileName,attr,omitempty"` // Storage profile name
+	Version                string    `xml:"version,attr,omitempty"`            // Storage profile name
+	LastSuccessfulSync     string    `xml:"lastSuccessfulSync,attr,omitempty"` // Date of last successful sync
+	Link                   *Link     `xml:"Link,omitempty"`
+	Metadata               *Metadata `xml:"Metadata,omitempty"`
+	NumberOfShadowVMs      int       `xml:"numberOfShadowVMs,attr,omitempty"`
```

Comment:
1. Why are we adding these fields?
2. Are they supported in every version?
3. Have you run a test to make sure the addition don't break existing code?

Reply:
I didn't find these fields in [docs] (https://code.vmware.com/apis/722/vmware-cloud-director/doc/doc/types/QueryResultVAppTemplateRecordType.html) but our vcd instance returns them. This is very strange and I don't know why.